mynav.cc

```diff
@@ -114,59 +95,45 @@ mynav_rd_line(const QString& line)
     }
   }
 
-  track_add_wpt(mynav_track, wpt);
+  track_add_wpt(track, wpt);
 }
 
+/***************************************************************************
+ *              entry points called by gpsbabel main process               *
+ ***************************************************************************/
 
-//***************************************************************************
-//           global callbacks called by gpsbabel main process
-//***************************************************************************
-
-static void
-mynav_rd_init(const QString& fname)
+void
+MyNavFormat::rd_init(const QString& fname)
 {
-  fin = gbfopen(fname, "rb", MYNAME);
-  mynav_track = route_head_alloc();
-  track_add_head(mynav_track);
+  read_fname = fname;
 }
 
-static void
-mynav_rd_deinit()
+void
+MyNavFormat::rd_deinit()
 {
-  gbfclose(fin);
+  read_fname.clear();
 }
 
-static void
-mynav_rd()
+void
+MyNavFormat::read()
 {
-  QString buff;
+  gpsbabel::File file(read_fname);
+  file.open(QIODevice::ReadOnly);
 
-  while ((buff = gbfgetstr(fin)), !buff.isNull()) {
-    buff = buff.trimmed();
-    if ((buff.isEmpty()) || (buff[0] == '#')) {
+  route_head* track = route_head_alloc();
+  track_add_head(track);
+
+  QTextStream stream(&file);
+  stream.setCodec("ASCII");
```

Comment:
Check out src/core/textstream.h, src/core/textstream.cc.  This can slightly simplify open and close.  It also has improved support for setting up a codec.  It will spit out all the known codecs if it doesn't find what you asked for.  I think you want US-ASCII, which is the preferred MIME name [https://www.iana.org/assignments/character-sets/character-sets.xhtml](url).  I don't think ASCII is a known alias.  Qt doesn't guarantee a US-ASCII codec so we provide our own in src/core to guarantee it is available on all platforms.

Reply:
Not sure where that "ASCII" came from. I will take a look what codec works best for this format.

Reply:
Actually "ASCII" does seem to be a valid alias. After textstream.setCodec("ASCII"), textstream.codec()->name() does return "US-ASCII".